### ISSUE TITLE
Possible conflict of help command parameters and getter

### ISSUE THREAD

**Issue by samdark** [yiisoft/yii2#12985]
See https://github.com/yiisoft/yii2/commit/cffb3832d19d7101e99a1cf30ec22caa17aadf30#commitcomment-19798475

**Comment by samdark** [comment]
Re-checked it. It is unlikely to cause any issues because both method and property are used directly.

**Comment by rob006** [comment]
@samdark This creates glitches in documentation: http://www.yiiframework.com/doc-2.0/yii-console-controller.html#$help-detail

We could at least fix this in 2.1 by renaming `getHelp()` to `getHelpInfo()`

**Comment by rob006** [comment]
There is more issues like that, so this probably will be easier to fix in `apidoc` extension...

Search `conflicts with a defined getter` in https://github.com/rob006/yii2-doc/blob/17aa8f7def035d9403db7f07fe454e7bcd0b183e/errors.txt

**Comment by samdark** [comment]
@cebe what do you think?

**Comment by cebe** [comment]
I think we should avoid conflicting methods and properties so a rename in 2.1 should be made.
We could also fix the apidoc but it is still confusing for users so I'd prefer not introducing such cases in the first place. I'd also like to make the error checking part of the travis build so we spot these directly when making a change.

**Comment by SamMousa** [comment]
We could create a test for this framework wide:
1. Iterate over all classes.
2. Reflect over their properties and methods.
3. Fail the test if we find a collision.

Not sure what suite that would belong in, but it makes sense to have these kinds of tests.

**Comment by rob006** [comment]
It is better to integrate documentation generation into tests - apidoc could detect much more bugs than just getters collision.

**Comment by SamMousa** [comment]
Yes, I'm thinking more broadly than just documentation generation though.
Regardless of documentation, getter collision is an issue even if we could properly document it.

**Comment by rob006** [comment]
Apidoc can detect getter collisions, so there is no point to create separate tool for that.

**Comment by SamMousa** [comment]
What about tests like usage of super globals? (Which you would want to limit only certain files via some whitelist)

**Comment by samdark** [comment]
Good idea as well.

**Comment by cebe** [comment]
> Apidoc can detect getter collisions, so there is no point to create separate tool for that.

it already does that: https://github.com/yiisoft/yii2-apidoc/blob/f66d08f2b8f6ed835f1fa92087c1594e1d3677f7/models/Context.php#L385 We just need to integrate that into CI.
